v23.1/connect-to-a-changefeed-kafka-sink-with-oauth-using-okta.md

```diff
@@ -0,0 +1,291 @@
+export OAUTH_LOGIN_SCOPE="kafka"
+export OAUTH_AUTHORIZATION="Basic {base64-encoded client credential value}"
+export OAUTH_INTROSPECT_SERVER="{Okta developer account URL}"
+export OAUTH_INTROSPECT_ENDPOINT="/oauth2/default/v1/introspect"
+export OAUTH_INTROSPECT_AUTHORIZATION="Basic {base64-encoded client credential value}"
+
+export CLASSPATH="{absolute path to kafka-oauth}/target/*"
+export KAFKA_OPTS="-Djava.security.auth.login.config={absolute path to kafka_2.13-2.8.2}/config/kafka_server_jaas.conf"
+~~~
+
+## Step 6. Start your Kafka server
+
+Ensure you're in the `kafka_2.13-2.8.2` directory, and then run the following:
+
+1. Start the Zookeeper server in one terminal:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    ./bin/zookeeper-server-start.sh ./config/zookeeper.properties
+    ~~~
+
+1. Start the Kafka server in another terminal:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    ./bin/kafka-server-start.sh ./config/server.properties
+    ~~~
+
+1. In another separate terminal window, create a topic to consume:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    ./bin/kafka-topics.sh --bootstrap-server localhost:9092 --create --topic vehicles
+    ~~~
+
+    To consume messages for this topic, run:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    ./bin/kafka-console-consumer.sh --bootstrap-server localhost:9092 --topic vehicles
+    ~~~
+
+## Step 7. Create a changefeed
+
+In this step, you will create a changefeed authenticating with Okta.
+
+The Kafka URI must follow this format:
+
+~~~
+'kafka://{kafka cluster address}:9093?topic_name={vehicles}&sasl_client_id={your client ID}&sasl_client_secret={your base64-encoded client secret}&sasl_enabled=true&sasl_mechanism=OAUTHBEARER&sasl_token_url={your url-encoded Okta developer token URL}'
+~~~
+
+Note the following:
+
+- You can include your client ID directly (without any encoding).
+- You must [base64 encode](https://www.base64encode.org/) your client secret.
+- You must [URL encode](https://www.urlencoder.org/) your Okta developer token URL, e.g., `https://dev-12345678.okta.com/oauth2/default/v1/token` will encode to `https%3A%2F%2Fdev-12345678.okta.com%2Foauth2%2Fdefault%2Fv1%2Ftoken`.
+
+Since this is a long URI to run, you can create an [external connection](create-external-connection.html) to represent this URI:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE EXTERNAL CONNECTION kafka_oauth AS 'kafka://{kafka cluster address}:9093?topic_name={vehicles}&sasl_client_id={your client ID}&sasl_client_secret={your base64-encoded client secret}&sasl_enabled=true&sasl_mechanism=OAUTHBEARER&sasl_token_url={your url-encoded Okta developer token URL}';
+~~~
+
+Create a changefeed that will emit messages to the topic consumer:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE CHANGEFEED FOR TABLE movr.vehicles INTO 'external://kafka_oauth' WITH updated, resolved;
+~~~
+~~~
```

Comment:
The last code block and where its content appears isn't really explained. After you run `CREATE CHANGEFEED`, is there no substantial response in the SQL shell, and then here, you're representing the changefeed data that is being sent to Kafka? A one-liner of some kind between the code blocks should be sufficient.

Reply:
Add explanation here